### PR TITLE
Suppress "no sources" warning when the target has dependencies

### PR DESCRIPTION
**Pull request by jsorge**
### Short description 📝

If there is a target which contains no source files, but has dependencies (such as dynamic target wrappers for static products) then Tuist produces a warning saying `The target {TargetName} doesn't contain source files.` However for these wrapper types there shouldn't be any source files.

This change suppresses that message for targets which don't have sources, but do have dependencies.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

**Review by ferologics — APPROVED** [review]
Thanks for this @jsorge! I look forward to saying goodbye to these 🚀
image

**Comment by apps4everyone** [comment]
Great job, maybe also if the target has actions like a `LintTarget`?

Screenshot 2021-04-29 at 22 59 46

**Comment by jsorge** [comment]
@apps4everyone I'm not sure if I understand. Would that be an aggregate target?

**Comment by apps4everyone** [comment]
> @apps4everyone I'm not sure if I understand. Would that be an aggregate target?

yes, they are basically empty target's (Lint, LintStrict, LintAnalytics) only with an action (SwiftLint.sh). Those action triggers the lint from Xcode.

Screenshot 2021-04-29 at 23 12 56

**Comment by jsorge** [comment]
Ah I see! Added a check for that as well @apps4everyone 🙂

File: Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift [hunk 1]
```diff
@@ -71,6 +71,24 @@ final class TargetLinterTests: TuistUnitTestCase {
         XCTContainsLintingIssue(got, LintingIssue(reason: "The target \(target.name) doesn't contain source files.", severity: .warning))
     }
 
+    func test_lint_when_target_no_source_files_but_has_dependency() {
+        let target = Target.test(sources: [], dependencies: [
+            TargetDependency.sdk(name: "libc++.tbd", status: .optional),
+        ])
```
Comment (kwridan) [inline Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift:77]: minor: 

I wonder, should it perhaps be `let hasNoDependencies = target.dependencies.filter(notSDKDependencies).isEmpty` - would there be a valid use case of a target with system dependencies only with no source code or build scripts ?

**Review by kwridan — APPROVED** [review]
Thanks @jsorge

File: Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift (hunk 1 above)
Comment (jsorge) [reply Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift:77]: That's an interesting edge case for sure. I'm not sure why anyone would do that, but if they did I think we'd probably want a different warning message. `Target has no sources and is only dependent on system frameworks` or something like that.

My gut feeling is that is such a rare case that I'm not sure if it's worth calling out though. It's possible that someone may have created a fresh target that way and they're on their way to adding sources, so the message would be more nuisance than help.

**Comment by luispadron** [comment]
What if we just removed this warning all together? Or moved into the `--verbose`? I fear that the number of times we don't want to show this will only grow and warnings / errors should be deterministic IMO

**Comment by apps4everyone** [comment]
> What if we just removed this warning all together? Or moved into the `--verbose`? I fear that the number of times we don't want to show this will only grow and warnings / errors should be deterministic IMO

`--verbose` would be great, also the `The bundle identifier 'XYZ' is being used by multiple targets...
